File: desktop/src/app/scenariolist/company.request.ts
/**
 * This class defines a model for requests to create companies in TraMS.
 */
export class CompanyRequest {

    private name: string;
    private startingBalance: number;
    private playerName: string;
    private startingTime: string;
    private scenarioName: string;
    private difficultyLevel: string;

    /**
     * Construct a new company request which contains the supplied data.
     * @param name the name of the company
     * @param startingBalance the starting balance of the company
     * @param playerName the player name who will run the company
     * @param startingTime the starting date and time for playing this company
     * @param scenarioName the name of the scenario that the player will play with this company
     * @param difficultyLevel the level of difficulty that the player wants to play
     */

    constructor( name: string, startingBalance: number, playerName: string, startingTime: string, scenarioName: string,
             difficultyLevel: string ) {
        this.name = name;
        this.startingBalance = startingBalance;
        this.playerName = playerName;
        this.startingTime = startingTime;
        this.scenarioName = scenarioName;
        this.difficultyLevel = difficultyLevel;
    }

}

File: server/src/main/java/de/davelee/trams/server/request/CompanyRequest.java
package de.davelee.trams.server.request;

/**
 * This class is part of the TraMS Server REST API. It represents a request to add the following company to the server
 * containing name, playerName and starting balance and time.
 * @author Dave Lee
 */
public class CompanyRequest {

    /**
     * The name of the company to request.
     */
    private String name;

    /**
     * The starting balance of this company.
     */
    private double startingBalance;

    /**
     * The player name for the company.
     */
    private String playerName;

    /**
     * The starting time for this company.
     */
    private String startingTime;

    /**
     * The scenario which this company was generated for (can be empty).
     */
    private String scenarioName;

    /**
     * The difficulty level which this company should be run at (can be EASY, MEDIUM or HARD)
     */
    private String difficultyLevel;

    public CompanyRequest() {
    }

    public CompanyRequest(String name, double startingBalance, String playerName, String startingTime, String scenarioName, String difficultyLevel) {
        this.name = name;
        this.startingBalance = startingBalance;
        this.playerName = playerName;
        this.startingTime = startingTime;
        this.scenarioName = scenarioName;
        this.difficultyLevel = difficultyLevel;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getStartingBalance() {
        return startingBalance;
    }

    public void setStartingBalance(double startingBalance) {
        this.startingBalance = startingBalance;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String playerName) {
        this.playerName = playerName;
    }

    public String getStartingTime() {
        return startingTime;
    }

    public void setStartingTime(String startingTime) {
        this.startingTime = startingTime;
    }

    public String getScenarioName() {
        return scenarioName;
    }

    public void setScenarioName(String scenarioName) {
        this.scenarioName = scenarioName;
    }

    public String getDifficultyLevel() {
        return difficultyLevel;
    }

    public void setDifficultyLevel(String difficultyLevel) {
        this.difficultyLevel = difficultyLevel;
    }

    @Override
    public String toString() {
        return "CompanyRequest{" +
                "name='" + name + '\'' +
                ", startingBalance=" + startingBalance +
                ", playerName='" + playerName + '\'' +
                ", startingTime='" + startingTime + '\'' +
                ", scenarioName='" + scenarioName + '\'' +
                ", difficultyLevel='" + difficultyLevel + '\'' +
                '}';
    }
}
